type_description_interfaces/msg/FieldType.msg

```diff
@@ -0,0 +1,149 @@
+# Represents the type of a field and related meta-data.
+
+# A constant for each type supported according to:
+#   http://design.ros2.org/articles/legacy_interface_definition.html
+# and:
+#   http://design.ros2.org/articles/idl_interface_definition.html
+# Order is loosely coupled to the order of appearance in the IDL 4.2 spec:
+#  https://www.omg.org/spec/IDL/4.2
+
+# Layout of constants across the 0-255 decimal values in the uint8:
+#
+# - 000-031: Primitive types, strings, and reserved space for future primitive types
+# - 032-063: Fixed sized array of primitive and string types
+# - 064-095: Bounded Sequences of primitive and string types
+# - 096-127: Unbounded Sequences of primitive and string types
+# - 128-255: Reserved space for future array/sequence-like types
+
+# Nested type defined in other .msg/.idl files.
+uint8 FIELD_TYPE_NESTED_TYPE = 0
+
+# Basic Types
+# Integer Types
+uint8 FIELD_TYPE_INT8 = 1
```

Comment:
Note: `rosidl.parser` has the categories of types, `[UN]SIGNED_NONEXPLICIT_INTEGER_TYPES` (containing (`unsigned`) `short`, `long`, `long long`) that are not represented here.
This could be an issue for translating from IDL->TypeDescription, but not the reverse.

Reply:
These messages (parts of TypeDescription.msg) are meant to represent what we can use in ROS 2, and since we have a declaration of what the values in msg translate to in idl, I think that we'd use the same matrix here.

Now if someone uses one of these other types in an .idl file, and we want to create a TypeDescription.msg from that, again I would say we use the matrix for converting between .msg and .idl, and if the type used in the .idl is outside of that, we abort saying we don't support it.